libs/common/src/vault/services/fido2/fido2-authenticator.service.ts

```diff
@@ -136,6 +136,14 @@ export class Fido2AuthenticatorService implements Fido2AuthenticatorServiceAbstr
         cipher = await encrypted.decrypt(
           await this.cipherService.getKeyForCipherKeyDecryption(encrypted)
         );
+
+        if (cipher.reprompt !== 0 && !userVerified) {
+          this.logService?.warning(
+            `[Fido2Authenticator] Aborting because user verification was unsuccessful.`
+          );
+          throw new Fido2AutenticatorError(Fido2AutenticatorErrorCode.NotAllowed);
+        }
+
```

Comment:
suggestion: consider merging this with the existing user verification check from line 125-129
https://github.com/bitwarden/clients/blob/dd7300fc5316add7f8f47a469c9d0372d9f0eefe/libs/common/src/vault/services/fido2/fido2-authenticator.service.ts#L125-L129

```suggestion

        if (!userVerified && (params.requireUserVerification || cipher.reprompt !== 0)) {
          this.logService?.warning(
            `[Fido2Authenticator] Aborting because user verification was unsuccessful.`
          );
          throw new Fido2AutenticatorError(Fido2AutenticatorErrorCode.NotAllowed);
        }

```